osutil/context.go

```diff
@@ -56,16 +58,24 @@ func RunWithContext(ctx context.Context, cmd *exec.Cmd) error {
 		return err
 	}
 
+	var ctxDone uint32
 	waitDone := make(chan struct{})
 	go func() {
 		select {
 		case <-ctx.Done():
+			atomic.StoreUint32(&ctxDone, 1)
 			cmd.Process.Kill()
 		case <-waitDone:
 		}
 	}()
 
 	err := cmd.Wait()
 	close(waitDone)
+	if atomic.LoadUint32(&ctxDone) != 0 {
+		// do one last check to make sure the error from Wait is what we expect from Kill
+		if err, ok := err.(*exec.ExitError); ok && err.ProcessState.Sys() == syscall.WaitStatus(0x9) {
```

Comment:
Bit more verbose:
```
if ws, ok := err.ProcessState.Sys().(syscall.WaitStatus); ok && ws.Signal() == syscall.SIGKILL {
   return ctx.Err()
}
```
Although looking at https://golang.org/src/syscall/syscall_linux.go#L194 your version is also correct.